samples/AppConfiguration/Generated/AppConfigurationRestClient.cs

```diff
@@ -24,14 +23,19 @@ internal partial class AppConfigurationRestClient
         private readonly string _syncToken;
         private readonly string _apiVersion;
 
+        /// <summary> The ClientDiagnostics is used to provide tracing support for the client library. </summary>
+        internal ClientDiagnostics ClientDiagnostics { get; }
+
         /// <summary> Initializes a new instance of AppConfigurationRestClient. </summary>
+        /// <param name="clientDiagnostics"> The handler for diagnostic messaging in the client. </param>
         /// <param name="pipeline"> The HTTP pipeline for sending and receiving REST requests and responses. </param>
         /// <param name="endpoint"> The endpoint of the App Configuration instance to send requests to. </param>
         /// <param name="syncToken"> Used to guarantee real-time consistency between requests. </param>
         /// <param name="apiVersion"> Api Version. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="pipeline"/>, <paramref name="endpoint"/> or <paramref name="apiVersion"/> is null. </exception>
-        public AppConfigurationRestClient(HttpPipeline pipeline, string endpoint, string syncToken = null, string apiVersion = "1.0")
+        /// <exception cref="ArgumentNullException"> <paramref name="clientDiagnostics"/>, <paramref name="pipeline"/>, <paramref name="endpoint"/> or <paramref name="apiVersion"/> is null. </exception>
+        public AppConfigurationRestClient(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, string endpoint, string syncToken = null, string apiVersion = "1.0")
         {
+            ClientDiagnostics = clientDiagnostics ?? throw new ArgumentNullException(nameof(clientDiagnostics));
```

Comment:
@AlexanderSher why are we reverting these changes?

Reply:
`ClientDiagnostics` type has a virtual method that can be overridden: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/ClientDiagnostics.cs#L52

Reply:
If `ExtractFailureContent` is overridden in derived type, there is no way to use it when we call `RequestFailedException` constructor directly.

Reply:
Would you mind adding this to the PR description and also linking the issue that will resolve this in the future?

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/27490